### DISCUSSION TITLE
FHIR Resources to represent Supply Chain

### DISCUSSION THREAD

**Discussion by roywanyaga** [opensrp/fhircore#2092] [General]
We will be tracking inventory change events using the [FHIR Observation resource](https://www.hl7.org/fhir/observation.html)

On a high level these [Observations](https://www.hl7.org/fhir/observation.html) denote the following aspects of the inventory change events:

- The type of the stock commodity affected by the event. Denoted by a reference to the [Group Resource](https://www.hl7.org/fhir/group.html) representing the stock commodity type.
- The category of stock inventory event. The possible categories include:
    1. Status listing (provision of the current quantity such as during physical count or stock adjustments). Represented by codeableConcept coding:
     `
   "category":  [
    {
      "coding": [
        {
          "system": "http://snomed.info/sct",
          "code": "386452003",
          "display": "Supply management"
        }
      ]
    },
    {
      "coding":  [
        {
          "system": "http://hl7.org/fhir/inventoryreport-counttype",
          "code": "snapshot",
          "display": "Snapshot"
        }
      ],
      "text": "Snapshot"
    }
  ]
`
    2. Reception (during stock supply or addition through other means). Represented by codeableConcept coding:
     `
 "category":  [
    {
      "coding": [
        {
          "system": "http://snomed.info/sct",
          "code": "386452003",
          "display": "Supply management"
        }
      ]
    },
    {
      "coding":  [
        {
          "system": "http://hl7.org/fhir/inventoryreport-counttype",
          "code": "addition",
          "display": "Addition"
        }
      ],
      "text": "Addition"
    }
  ]
`
    3. Depletion (Through consumption, issuance to patents, use during service delivery, damage, expiry, or other means). Represented by codeableConcept coding:
         `
  "category":  [
    {
      "coding": [
        {
          "system": "http://snomed.info/sct",
          "code": "386452003",
          "display": "Supply management"
        }
      ]
    },
    {
      "coding":  [
        {
          "system": "http://hl7.org/fhir/inventoryreport-counttype",
          "code": "subtraction",
          "display": "Subtraction"
        }
      ],
      "text": "Subtraction"
    }
  ]
`

- The number of stock items added, subtracted, or are present
- The type of the stock-related event (such as physical count, stock supply, patient visit, donation, expiry, damage, etc.). These are detailed in the table below.
- The date and time of the event
- The updated stock count for the commodity type as a result of the event

For each of the commodity types, multiple [Observations](https://www.hl7.org/fhir/observation.html) will be created depending on the type of stock inventory events. 

The different possible stock inventory events, their codes, categories, questionnaire fields, and condition for relevance are provided below:

During the supervisor visit (physical count and restock)
****
| Type | Code | Category |
|--- | --- | --- |
| System Balance before restocking | balance-before-restock | snapshot |
| Stock on Hand after Physical Inventory Count | physical-count-soh | snapshot |
| Adjustment due to over-reporting | over-reporting | subtraction  |
| Adjustment due to under-reporting | under-reporting | addition |
| Adjustment due to expiry | expiry | subtraction |
| Adjustment due to damage | damage | subtraction |
| Adjustment due to donation | donation| addition |
| Stock supply/resupply | restocked | addition |
| Balance after adjustment and restocking | balance-after-restock | snapshot |

During the patient visits and service delivery
****
| Type | Code | Category |
|--- | --- | --- |
| Stock use/depletion during patient visit | consumption | subtraction |

**Comment by pld** [comment]
Can you add more info on how the over/under reporting works?
